### ISSUE TITLE
Overabundance of mulattos

### ISSUE THREAD

**Issue by devolution79** [We-the-People-civ4col-mod/Mod#1028]
Several players have noted that mulattos seem to be overrepresented after population growth.
I had a look at the function that is responsible for determining the ethnicity CvCity::bestGrowthUnitClass and I noticed that the random selection is biased due to the order. Since the random draw for mulattoes are the last operation, this results in significantly higher odds for mulattoes spawning.

I asked ChatGTP to simulate (1,000,000  trials) / compute the odds and these were the result:

Ethnicity European selected 98826 times (9.88%)
Ethnicity Indio selected 98556 times (9.86%)
Ethnicity African selected 99457 times (9.95%)
Ethnicity Mestizzo selected 99500 times (9.95%)
Ethnicity Mulatto selected 402661 times (40.27%)

As we can see, the probability of a mulatto spawning is far higher than what is probably intended.

Thus I've reworked the function to be "fair" in the sense that the probability of any given ethnicity spawning after growth is now computed as an unbiased weighted probability based on the relative proportion of ethnicity candidates.
In other words, given an equal number of each ethnicity (currently 5 variants), the odds of any particular spawning is 1/5

https://github.com/We-the-People-civ4col-mod/Mod/tree/unbiased_bestGrowthUnitClass

@raystuttgart  (and others): What do you think of this change ?

**Comment by MrZorG33** [comment]
with an equal number of “types” of colonists, the changes seem correct to me. question: what rule will work if there is an unequal number of “species” of colonists? in other words: will, say, “mestizo” have an advantage over “еuropeans” with a ratio of 3/2?

**Comment by devolution79** [comment]
@MrZorG33 It will work as you'd expect. The outcome for that particular situation would be 50% more mestizos (on average in the long run)

**Comment by raystuttgart** [comment]
My idea was basically to match Population diversity to Growth.
If there was accidently something unbalanced, go ahead and fix it.

Never intended to have more excessive amounts of Mulattos if the population rates would not explain it mathematially.
I did not want to have an "illegal immigration feature" or anything like that. :)

**Comment by devolution79** [comment]
Merged!
